go-concourse/concourse/internal/connection_test.go

```diff
@@ -355,7 +357,7 @@ var _ = Describe("ATC Connection", func() {
 				BeforeEach(func() {
 					atcServer = ghttp.NewServer()
 
-					connection = NewConnection(atcServer.URL(), nil, tracing)
+					agent = NewHTTPAgent(atcServer.URL(), nil, tracing)
```

Comment:
How come we're testing the http_agent in the connection_test file? Should we have a separate suite for http_agent_test? 

Overall a little confused why we have both the connection and agent? Should we just refactor the connection instead of deprecating it?

Reply:
We want to use `HTTPAgent` everywhere but the diff on this refactor is already really large and this would've dramatically increased the scope. We're holding ourselves back and only making this one change while indicating where we want to see this change end up, with `connection` deleted and replaced by `HTTPAgent`

Reply:
It would probably keep the diff small if we refactored `connection` to look like `agent` and then continued using `connection` everywhere.

Reply:
I disagree that the diff would be smaller  and we would still end up with a larger refactor (maybe I'm misunderstanding what you're suggesting?). This is how I'm interperting your suggestion and why I think it would result in a larger diff.

`connection` does two things: 1) send HTTP requests and 2) parse the response. 
`agent` does one thing 1) send HTTP requests. 
The response parsing still needs to happen though, who's responsible for that now? The caller has to do this now (and was already doing part of this but in a non-extensible way). This extra responsibility is what allowed us to return better error messages when `--team` is used.

For the one case (`FindTeam`) currently using `agent` you can see the response parsing happening here: 

https://github.com/concourse/concourse/blob/587483424a47f3eebb2ed8e861687b60cccb9ec7/go-concourse/concourse/teams.go#L122-L145

If we were to to switch everything over we'd have to change this section:
https://github.com/concourse/concourse/blob/587483424a47f3eebb2ed8e861687b60cccb9ec7/go-concourse/concourse/teams.go#L84-L97

 in every area that uses `connection` which is currently 71 places, every API call:
```
ag 'connection.Send' ./go-concourse/ | wc -l
      71
```

I think this will result in a much larger diff and will take a good chunk of time to refactor. I think this can be it's own story, finishing the refactor that was started in this story. wdyt?

Reply:
yeah ok, I didn't realize that the response handling would be that different.

Reply:
Made an issue to complete this refactor: #5214